catalyst-toolbox/src/bin/cli/rewards/full/config.rs

```diff
@@ -0,0 +1,54 @@
+use std::path::PathBuf;
+
+use rust_decimal::Decimal;
+use serde::Deserialize;
+
+use crate::cli::rewards::community_advisors::{FundSettingOpt, ProposalRewardsSlotsOpt};
+
+#[derive(Debug, Deserialize)]
+pub(super) struct Config {
+    pub(super) inputs: Inputs,
+    pub(super) outputs: Outputs,
+    pub(super) params: Params,
+}
+
+#[derive(Debug, Deserialize)]
+pub(super) struct Inputs {
+    pub(super) block_file: PathBuf,
+    pub(super) snapshot_path: PathBuf,
+    pub(super) vote_count_path: PathBuf,
+    pub(super) reviews_csv: PathBuf,
+    pub(super) assessments_path: PathBuf, // is assessments the same as reviews?
+    pub(super) proposal_bonus_output: Option<PathBuf>,
+    pub(super) approved_proposals_path: PathBuf,
+    pub(super) proposer_script_path: PathBuf,
+    pub(super) active_voteplans: PathBuf,
+    pub(super) challenges: PathBuf,
+    pub(super) proposals_path: PathBuf,
+}
+
+#[derive(Debug, Deserialize)]
+pub(super) struct Outputs {
+    pub(super) veterans_rewards_output: PathBuf,
+    pub(super) ca_rewards_output: PathBuf,
+    pub(super) proposer_rewards_output: PathBuf,
+}
+
+#[derive(Debug, Deserialize)]
+pub(super) struct Params {
+    pub(super) registration_threshold: u64,
+    pub(super) vote_threshold: u64,
+    pub(super) total_rewards: u64,
```

Comment:
I think this should be separated for each rewards type

Reply:
Is this referring to the conversation in the ticket? I read through it but wasn't sure what the implications were for this PR. Is the "reward type" refrerring to ada/lovelace/dollars or something else?

Reply:
Nono, I mean that each rewards (i.e. proposer, voters, vca, ca) will likely have a different value for `total_rewards`, e.g. 100 ADA for voter rewards, 50 for vCA

Reply:
@cameron1024  Are we going to split this rewards in this PR ? Or it should be a part of another ticket ?